packages/react-data-grid/src/Row.tsx

```diff
@@ -42,38 +42,39 @@ export default class Row<R> extends React.Component<IRowRendererProps<R>> {
       onRowSelectionChange,
       row,
       scrollLeft,
-      isSummaryRow
+      isSummaryRow,
+      ...props
     } = this.props;
     const Renderer = this.props.cellRenderer!;
-    const cellElements = [];
 
-    for (let colIdx = 0; colIdx <= colOverscanEndIdx; colIdx++) {
-      const column = columns[colIdx];
-      const colIsFrozen = isFrozen(column);
-
-      if (colIdx < colOverscanStartIdx && !colIsFrozen) continue;
-
-      const { key } = column;
-
-      cellElements.push(
-        <Renderer
-          key={key as string} // FIXME: fix key type
-          idx={colIdx}
-          rowIdx={idx}
-          column={column}
-          lastFrozenColumnIndex={lastFrozenColumnIndex}
-          cellMetaData={cellMetaData}
-          rowData={row}
-          expandableOptions={this.getExpandableOptions(key)}
-          scrollLeft={colIsFrozen && typeof scrollLeft === 'number' ? scrollLeft : undefined}
-          isRowSelected={isRowSelected}
-          onRowSelectionChange={onRowSelectionChange}
-          isSummaryRow={isSummaryRow}
-        />
-      );
-    }
-
-    return cellElements;
+    return getViewportColumns(columns, colOverscanStartIdx, colOverscanEndIdx)
```

Comment:
Don't we pre-filter the visible columns now? We should only pass the visible columns so we don't have to call `getViewportColumns` in each Row component, wdyt?

Reply:
I thought about it but decided to keep this PR simple. `viewportColumns` is recomputed when `scrollLeft` is changed so it may break memoization. Let's revisit it in the future